src/exchange/exchange.ts

```diff
@@ -20,7 +20,10 @@ export function createYesterdayPrice$(
 ): Observable<BigNumber | undefined> {
   return tradeHistory$.pipe(
     map(trades => {
-      const borderline = moment().subtract(1, 'days').toDate();
+      if (!trades[0]) {
+        return undefined;
```

Comment:
I remember there was this idea that you never accept nullable values and you never return such. For me returning an undefined could trigger some hidden bugs ( correct me if I'm wrong  or I miss smth )

Reply:
In this case the idea does not apply, because it is not always possible to know yesterdayPrice (eg. when no trades at all). So no other option here.
Generally TypeScript helps you find such bugs. Note that all the observables are typed. The TradingPairView also handles all the cases properly.